### PR TITLE
Pending until discover status up

### PR DESCRIPTION
**Pull request by Flanker32**
Pending until discover status up, for issue #797

### CODE REVIEW

File: azure-spring-cloud-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/DeployMojo.java [hunk 1]
```diff
@@ -170,8 +170,13 @@ protected boolean isDeploymentDone(DeploymentResourceInner deploymentResource) {
         if (deploymentResource.properties().status() == DeploymentResourceStatus.PROCESSING) {
             return false;
         }
-        return !deploymentResource.properties().instances().stream()
-                .anyMatch(instance -> instance.status().equalsIgnoreCase("waiting") || instance.status().equalsIgnoreCase("pending"));
+        final String finalDiscoverStatus = deploymentResource.properties().active() ? "UP" : "OUT_OF_SERVICE";
```
Comment (andxu) [inline azure-spring-cloud-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/DeployMojo.java]: active() returns Boolean

File: azure-spring-cloud-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/DeployMojo.java [hunk 2]
```diff
@@ -170,8 +170,13 @@ protected boolean isDeploymentDone(DeploymentResourceInner deploymentResource) {
...
+        final List<DeploymentInstance> instanceList = deploymentResource.properties().instances();
+        final boolean instanceStatus = !instanceList.stream().anyMatch(instance ->
+                instance.status().equalsIgnoreCase("waiting") || instance.status().equalsIgnoreCase("pending"));
```
Comment (andxu) [inline azure-spring-cloud-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/DeployMojo.java]: reverse order "waiting".equalsIgnoreCase(xxx)

File: azure-spring-cloud-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/DeployMojo.java [hunk 3]
```diff
@@ -170,8 +170,13 @@ protected boolean isDeploymentDone(DeploymentResourceInner deploymentResource) {
...
+        final boolean discoverStatus = instanceList.stream().allMatch(instance ->
+                instance.discoveryStatus().equalsIgnoreCase(finalDiscoverStatus));
```
Comment (andxu) [inline azure-spring-cloud-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/DeployMojo.java]: reverse order also

File: azure-spring-cloud-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/DeployMojo.java [hunk 4]
```diff
@@ -170,8 +170,13 @@ protected boolean isDeploymentDone(DeploymentResourceInner deploymentResource) {
         if (deploymentResource.properties().status() == DeploymentResourceStatus.PROCESSING) {
             return false;
         }
-        return !deploymentResource.properties().instances().stream()
-                .anyMatch(instance -> instance.status().equalsIgnoreCase("waiting") || instance.status().equalsIgnoreCase("pending"));
+        final String finalDiscoverStatus = deploymentResource.properties().active() ? "UP" : "OUT_OF_SERVICE";
+        final List<DeploymentInstance> instanceList = deploymentResource.properties().instances();
+        final boolean instanceStatus = !instanceList.stream().anyMatch(instance ->
```
Comment (andxu) [inline azure-spring-cloud-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/DeployMojo.java]: instanceStatus doesn't indicates it is a boolean, instanceStatus => instancePending

File: azure-spring-cloud-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/DeployMojo.java (hunk 3 above)
Comment (andxu) [reply azure-spring-cloud-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/DeployMojo.java]: discoverStatus -> discoverStatusFinalized

Comment (andxu) [reply azure-spring-cloud-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/DeployMojo.java]: allMatch ? all instances should be OUT_OF_SERVICE?

Comment (Flanker32) [reply azure-spring-cloud-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/DeployMojo.java]: yes, the final status of inactive deployment instances shoudl be `OUT_OF_SERVICE`

File: azure-spring-cloud-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/DeployMojo.java (hunk 1 above)
Comment (Flanker32) [reply azure-spring-cloud-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/DeployMojo.java]: Changed to use `BooleanUtils`

File: azure-spring-cloud-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/DeployMojo.java (hunk 2 above)
Comment (Flanker32) [reply azure-spring-cloud-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/DeployMojo.java]: Changed to `StringUtils.equalsIgnoreCases`

File: azure-spring-cloud-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/DeployMojo.java (hunk 4 above)
Comment (Flanker32) [reply azure-spring-cloud-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/DeployMojo.java]: Changed to `isInstanceDeployed`

**Pull request merged**